### PR TITLE
gadget,overlord: pass kernelRoot to install.Run()

### PR DESCRIPTION
**Pull request by mvo5**
This commit adds kernelRoot to install.Run() so that kernel
references can be resolved during install. No further functionality
is changed this is the first trivial split out of #9780 to make
the review easier.

### CODE REVIEW

File: gadget/install/install.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func deviceFromRole(lv *gadget.LaidOutVolume, role string) (device string, err e
 
 // Run bootstraps the partitions of a device, by either creating
 // missing ones or recreating installed ones.
-func Run(model gadget.Model, gadgetRoot, device string, options Options, observer gadget.ContentObserver) (*InstalledSystemSideData, error) {
+func Run(model gadget.Model, gadgetRoot, kernelRoot, device string, options Options, observer gadget.ContentObserver) (*InstalledSystemSideData, error) {
```
Comment (anonymouse64) [inline gadget/install/install.go:56]: I think there may be too many parameters here, can gadgetRoot and kernelRoot and things go into the Options (or maybe rename Options to something like Params, etc. ) ?

**Review by anonymouse64 — APPROVED** [review]
lgtm with one suggestion previously, but that's not a blocker

File: gadget/install/install.go (hunk 1 above)
Comment (bboozzoo) [reply gadget/install/install.go:56]: 6 params may be a bit too many indeed. Extending Options sounds ok to me.

**Review by bboozzoo — APPROVED** [review]
LGTM, assuming the suggestion about tweaking the parameters list of install.Run() is applied.

File: gadget/install/install.go (hunk 1 above)
Comment (pedronis) [reply gadget/install/install.go:56]: I think 6 is borderline, having Run(Options)  would also be a bit weird to me

Comment (mvo5) [reply gadget/install/install.go:56]: I added a TODO for myself to look at this for a followup

**Pull request merged**